### PR TITLE
Rename GraphObject & GraphList that's backwards-compatible

### LINKED ISSUE #320: [6.0] [Proposal] Rename GraphObject & GraphList
After discussing the Facebook PHP SDK a bit on the [domain-driven design discussion on the PHP Roundtable](https://www.phproundtable.com/episode/domain-driven-design-in-php), I realized that there are a few places in the SDK where we could better be in sync with Facebook nomenclature.

The two biggest DDD-eye-sores are the naming `GraphObject` & `GraphList`.

1. The `GraphObject` represents a "thing" on the Graph API. ["Things" in Graph are called "nodes"](https://developers.facebook.com/docs/graph-api/quickstart/v2.2#basics). We started going the right direction by renaming the containing folder to `GraphNodes` but let's go ahead and complete that by renaming `GraphObject` to `GraphNode`.
2. Relationships between nodes in Graph [are called "edges"](https://developers.facebook.com/docs/graph-api/quickstart/v2.2#basics).The `GraphList` represents one or more nodes returned by Graph from an edge. I recommend we rename `GraphList` to `GraphEdge`.

This change will disambiguate what these objects represent and reduce confusion when someone is reading through the Graph API and Facebook SDK documentation. :)

### PR DESCRIPTION
**Pull request by SammyK**
This will close #320.

Now that we're adopting SemVer we can use `MAJOR` versions to deprecate stuffs. Yay!

It looks like a big PR, but it's really just a tiny rename that touches many places in the codebase.

- `GraphObject` is now `GraphNode` & `GraphObject` is deprecated with the `@deprecated` flag.
- `GraphList` is now `GraphEdge` & `GraphList` is deprecated with the `@deprecated` flag.
- `GraphObjectFactory` is now `GraphNodeFactory` & `GraphObjectFactory` is deprecated with the `@deprecated` flag.
- `FacebookResponse::getGraphList()` has been `@deprecated` & renamed to `getGraphEdge()`
- `FacebookResponse::getGraphObject()` has been `@deprecated` & renamed to `getGraphNode()`

This shouldn't be a breaking change for anyone on 4.1 unless they've really overwritten deep functionality of the codebase.

I've still got to update the docs so don't merge yet. :)

### CODE REVIEW

**Comment by SammyK** [comment]
Docs updated!

Just realized that this is going to hardcore break @devmsh's PR #405, so you might want to merge his in first when he finalizes his changes. Then I'll go in an fix any conflicts on this PR before merging. :)

**Comment by SammyK** [comment]
Fixed conflicts with #405 so this should be good to go pending any feedback from @yguedidi :)

File: docs/FacebookRequest.fbmd [hunk 1]
```diff
@@ -51,9 +51,9 @@ try {
   exit;
 }
 
-$object = $response->getGraphObject();
+$node = $response->getGraphNode();
```
Comment (yguedidi) [inline docs/FacebookRequest.fbmd]: Why not `$graphNode` here too?

File: docs/GraphNode.fbmd [hunk 2]
```diff
@@ -25,8 +25,8 @@ $fb = new Facebook\Facebook(\* *\);
 // Returns a `Facebook\FacebookResponse` object
 $response = $fb->get('/something');
 
-// Get the base class GraphObject from the response
-$node = $response->getGraphObject();
+// Get the base class GraphNode from the response
+$node = $response->getGraphNode();
```
Comment (yguedidi) [inline docs/GraphNode.fbmd]: `$graphNode` herer too

File: docs/example_pagination_basic.fbmd [hunk 3]
```diff
@@ -32,7 +32,7 @@ try {
 }
 
 // Page 1
-$feed = $response->getGraphList();
+$feed = $response->getGraphEdge();
```
Comment (yguedidi) [inline docs/example_pagination_basic.fbmd]: `$feedEdge`?

File: tests/GraphNodes/GraphObjectFactoryTest.php [hunk 4]
```diff
@@ -235,203 +73,42 @@ public function testAGraphObjectWillBeCastAsAGraphObject()
...
+            'previous' => 'http://facebook/prev',
+          ],
         ]);
-
         $res = new FacebookResponse($this->request, $data);
 
         $factory = new GraphObjectFactory($res);
-        $graphObject = $factory->makeGraphObject();
-        $graphData = $graphObject->asArray();
+        $graphEdge = $factory->makeGraphList();
```
Comment (yguedidi) [inline tests/GraphNodes/GraphObjectFactoryTest.php]: `makeGraphEdge`?

File: tests/GraphNodes/GraphObjectFactoryTest.php [hunk 5]
```diff
@@ -235,203 +73,42 @@ public function testAGraphObjectWillBeCastAsAGraphObject()
...
+        $graphData = $graphEdge->asArray();
 
-        $this->assertInstanceOf('\Facebook\GraphNodes\GraphObject', $graphObject);
+        $this->assertInstanceOf('\Facebook\GraphNodes\GraphList', $graphEdge);
```
Comment (yguedidi) [inline tests/GraphNodes/GraphObjectFactoryTest.php]: `'\Facebook\GraphNodes\GraphEdge'`?

File: tests/GraphNodes/GraphObjectFactoryTest.php (hunk 4 above)
Comment (SammyK) [inline tests/GraphNodes/GraphObjectFactoryTest.php]: This is to show BC but renamed `$graphEdge` to `$graphList` to be more explicit what it's testing. :)

File: tests/GraphNodes/GraphObjectFactoryTest.php (hunk 5 above)
Comment (SammyK) [inline tests/GraphNodes/GraphObjectFactoryTest.php]: Also to show BC but renamed `$graphEdge` to `$graphList`.

**Comment by SammyK** [comment]
Thanks for the review @yguedidi! Done. :)

**Comment by SammyK** [comment]
Also just updated per [@yguedidi's suggestion](https://github.com/facebook/facebook-php-sdk-v4/issues/429#issuecomment-101688077). Renamed `getProperty()` to `getField()` and deprecated `getProperty()`.

**Comment by gfosco** [comment]
Awesome

**Pull request merged**